File: app/src/main/java/org/cuieney/videolife/presenter/EssayHomeDetailPresenter.java
package org.cuieney.videolife.presenter;

import org.cuieney.videolife.common.api.OpApiService;
import org.cuieney.videolife.common.base.RxPresenter;
import org.cuieney.videolife.common.utils.RxUtil;
import org.cuieney.videolife.entity.EssayDetailBean;
import org.cuieney.videolife.presenter.contract.EssayHomeDetailContract;

import javax.inject.Inject;

import rx.Subscription;
import rx.functions.Action1;

/**
 * Created by cuieney on 2017/5/17.
 */

public class EssayHomeDetailPresenter extends RxPresenter<EssayHomeDetailContract.View> implements EssayHomeDetailContract.Presenter {

    private OpApiService mRetrofitHelper;

    @Inject
    public EssayHomeDetailPresenter(OpApiService mRetrofitHelper) {
        this.mRetrofitHelper = mRetrofitHelper;
    }

    @Override
    public void getEssayDetailData(int id, int sourceId) {
        Subscription rxSubscription = mRetrofitHelper.getEssayDetail(id,sourceId)
                .compose(RxUtil.rxSchedulerHelper())
                .subscribe(essayDetailBean -> mView.showContent(essayDetailBean), throwable -> mView.error(throwable));
        addSubscrebe(rxSubscription);
    }
}

File: app/src/main/java/org/cuieney/videolife/kotlin/presenter/EssayHomeDetailPresenter.kt
package org.cuieney.videolife.kotlin.presenter

import org.cuieney.videolife.common.api.OpApiService
import org.cuieney.videolife.common.utils.RxUtil
import org.cuieney.videolife.entity.EssayDetailBean
import org.cuieney.videolife.kotlin.base.RxPresenter
import org.cuieney.videolife.kotlin.presenter.contract.EssayHomeDetailContract
import javax.inject.Inject

/**
 * Created by cuieney on 2017/5/23.
 */
class EssayHomeDetailPresenter @Inject
constructor(private val mRetrofitHelper: OpApiService) : RxPresenter(), EssayHomeDetailContract.Presenter {

    override fun getEssayDetailData(id: Int, sourceId: Int,mView:EssayHomeDetailContract.View) {
        val rxSubscription = mRetrofitHelper.getEssayDetail(id, sourceId)
                .compose(RxUtil.rxSchedulerHelper<EssayDetailBean>())
                .subscribe({ essayDetailBean -> mView.showContent(essayDetailBean) }) { throwable -> mView.error(throwable) }
        addSubscrebe(rxSubscription)
    }
}